nucliadb/src/nucliadb/writer/api/v1/knowledgebox.py

```diff
@@ -32,61 +38,101 @@
+        if not is_onprem_nucliadb():
+            raise HTTPException(
+                status_code=403,
+                detail="This endpoint is only available for onprem NucliaDB",
+            )
+        return await fun(*args, **kwargs)
 
+    return endpoint_wrapper
 
+
+@only_for_onprem
 @api.post(
     f"/{KBS_PREFIX}",
     status_code=201,
     summary="Create Knowledge Box",
-    response_model=KnowledgeBoxObj,
     tags=["Knowledge Boxes"],
     openapi_extra={"x-hidden-operation": True},
 )
 @requires(NucliaDBRoles.MANAGER)
 @version(1)
-async def create_kb(request: Request, item: KnowledgeBoxConfig):
-    ingest = get_ingest()
-    requestpb = KnowledgeBoxNew()
-    requestpb = parse_create_kb_request(item)
-    kbobj: NewKnowledgeBoxResponse = await ingest.NewKnowledgeBox(requestpb)  # type: ignore
-    if item.slug != "":
-        slug = item.slug
+async def create_kb(request: Request, item: KnowledgeBoxConfig) -> KnowledgeBoxObj:
+    driver = get_driver()
+    rollback_learning_config = None
+
+    kbid = KnowledgeBox.new_unique_kbid()
+
+    # Onprem KBs have to call learning proxy to create it's own configuration.
+    if item.learning_configuration:
+        user_learning_config = item.learning_configuration
     else:
-        slug = kbobj.uuid
-    if kbobj.status == KnowledgeBoxResponseStatus.OK:
-        return KnowledgeBoxObj(uuid=kbobj.uuid, slug=slug)
-    elif kbobj.status == KnowledgeBoxResponseStatus.CONFLICT:
+        logger.warning(
+            "No learning configuration provided. Default will be used.",
+            extra={"kbid": kbid},
+        )
+        # learning will choose the default values
+        user_learning_config = {}
+
+    # we rely on learning to return the updated configuration with defaults and
+    # any other needed values (e.g. matryoshka settings if available)
+    learning_config = await learning_proxy.set_configuration(kbid, config=user_learning_config)
+
+    # if KB creation fails, we'll have to delete its learning config
+    async def _rollback_learning_config(kbid: str):
+        try:
+            await learning_proxy.delete_configuration(kbid)
+        except Exception:
+            logger.warning(
+                "Could not rollback learning configuration",
+                exc_info=True,
+                extra={"kbid": kbid},
+            )
+            raise
+
+    rollback_learning_config = partial(_rollback_learning_config, kbid)
+
+    config = knowledgebox_pb2.KnowledgeBoxConfig(
+        title=item.title or "",
+        description=item.description or "",
+    )
+    semantic_model = learning_config.into_semantic_model_metadata()
+    release_channel = item.release_channel.to_pb() if item.release_channel is not None else None
+    try:
+        async with driver.transaction() as txn:
+            kbid, failed = await KnowledgeBox.create(
+                txn,
+                slug=item.slug or "",  # empty slugs will be changed on KB creation
```

Comment:
Should this param be optional then?

Reply:
Probably, I didn't touch too much the signature of `create` to keep it simpler, but IMO we shouldn't be passing `config` either and we should have something like:
```
async def create(
    kbid: str,
    *,
    slug: Optional[str] = None,
    title: Optional[str] = None,
    description: Optional[str] = None,
    semantic_model: knowledgebox_pb2.SemanticModelMetadata,
    release_channel: Optional[
        utils_pb2.ReleaseChannel.ValueType
    ] = utils_pb2.ReleaseChannel.STABLE,
) -> tuple[str, str]:
```
returning `(kbid, slug)`.

WDYT?